### PR TITLE
feat: adapted search frontend to backend

### PR DESCRIPTION
**Pull request by heydoyouknowme0**

### CODE REVIEW

File: server/typesense/index.ts [hunk 1]
```diff
@@ -6,14 +6,12 @@ const createTypesenseClient = () =>
...
-        ...(env.NODE_ENV === 'production'
-          ? { protocol: 'https' }
-          : { protocol: 'http' }),
+        host: env.NEXT_PUBLIC_TYPESENSE_HOST,
+        port: env.NEXT_PUBLIC_TYPESENSE_PORT,
+        protocol: 'http',
       },
     ],
-    apiKey: process.env.TYPESENSE_API_KEY!,
+    apiKey: env.NEXT_PUBLIC_TYPESENSE_API_KEY,
```
Comment (GetPsyched) [inline server/typesense/index.ts:14]: Huh, why would you expose the Typesense API key to the client? This is a security risk.

File: lib/env.ts [hunk 2]
```diff
@@ -19,16 +19,22 @@ export const env = createEnv({
     AWS_S3_REGION: z.string(),
     AWS_PUBLIC_S3_NAME: z.string(),
     AWS_PRIVATE_S3_NAME: z.string(),
-
-    TYPESENSE_HOST: z.string(),
-    TYPESENSE_PORT: z.coerce.number(),
-    TYPESENSE_API_KEY: z.string(),
```
Comment (GetPsyched) [inline lib/env.ts:25]: Env vars can remain in both server and client

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: server/typesense/index.ts (hunk 1 above)
Comment (heydoyouknowme0) [reply server/typesense/index.ts:14]: It just wasn't working without making it public, now that I think about it that shouldn't be required, I'll look that

File: server/typesense/utils.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+'use server';
```
[the remark this answers is not in the document]
Comment (heydoyouknowme0) [reply server/typesense/utils.ts:1]: Without it I was getting client env errors, randomly

**Pull request merged**